### PR TITLE
Feature to check whether a website is running or not

### PR DESCRIPTION
**Pull request by Wert1996**
Checks whether a web domain is up or not using isitup.org. Needs to learn 'Is xyz.com up', 'xyz.com status'.

### CODE REVIEW

File: modules/src/__init__.py [hunk 1]
```diff
@@ -21,6 +21,7 @@
     'url',
     'video',
     'weather',
+    'webcheck',
```
Comment (swapagarwal) [inline modules/src/__init__.py]: Can we rename it to ping module? It's more intuitive.

File: modules/src/webcheck.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+from templates.text import TextTemplate
+
+def process(input, entitites):
+    output = {}
+    try:
+        url = entities['url'][0]['value']
+        strip_list = ['http://','https://','/']
+        for item in strip_list:
+            url = url.strip(item)
+        page = requests.get('https://isitup.org/'+str(url))
```
Comment (swapagarwal) [inline modules/src/webcheck.py]: Pro Tip: See what happens if you append .json to the end of the URL..

**Comment by Wert1996** [comment]
Thanks for the feedback! Applied the proposed changes :)

File: modules/src/ping.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+def process(input, entitites):
+    output = {}
+    try:
+        url = entities['url'][0]['value']
+        orig_url = url
+        strip_list = ['http://','https://','/']
+        for item in strip_list:
+            url = url.strip(item)
+        page = requests.get('https://isitup.org/'+url+'.json')
+        page = json.loads(page.text)
```
Comment (swapagarwal) [inline modules/src/ping.py]: If you see the other modules, the general structure is:
r = requests.get(...)
data = r.json()

File: modules/src/ping.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    except:
+        error_message = 'There seems to be some problem'
+        error_message = '\nPlease ask me something else, like:'
+        error_message = '\n - is google.com up'
+        error_message = '\n - google.com status'
+        error_message = '\n - ping google.com'
+        output['error_msg'] = TextTemplate(error_message).get_message()
+        output['success'] = False
+    return output
+
```
Comment (swapagarwal) [inline modules/src/ping.py]: Remove extra line.

File: modules/src/ping.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        elif status == 2:
+            txt = orig_url + ' seems to be down!'
+        else :
+            txt = 'I need a valid domain to check!'
+        output['input'] = input
+        output['output'] = TextTemplate(txt).get_message()
+        output['success'] = True
+    except:
+        error_message = 'There seems to be some problem'
+        error_message = '\nPlease ask me something else, like:'
```
Comment (swapagarwal) [inline modules/src/ping.py]: It should be "+=". Check the below lines too.

File: modules/src/ping.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        url = entities['url'][0]['value']
+        orig_url = url
+        strip_list = ['http://','https://','/']
+        for item in strip_list:
+            url = url.strip(item)
+        page = requests.get('https://isitup.org/'+url+'.json')
+        page = json.loads(page.text)
+        status = page['status_code']
+        if status == 1:
+            txt = orig_url + ' is up!'
```
Comment (swapagarwal) [inline modules/src/ping.py]: Don't use original url here, as that might be misleading. Use the one that we passed in the api call.

File: modules/src/ping.py [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+        elif status == 2:
+            txt = orig_url + ' seems to be down!'
+        else :
```
Comment (swapagarwal) [inline modules/src/ping.py]: Do we know the status code here? If yes, then we should use that here instead. Any unexpected status codes should result in an error.

File: modules/src/ping.py [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+def process(input, entitites):
+    output = {}
+    try:
+        url = entities['url'][0]['value']
+        orig_url = url
+        strip_list = ['http://','https://','/']
+        for item in strip_list:
+            url = url.strip(item)
+        page = requests.get('https://isitup.org/'+url+'.json')
```
Comment (swapagarwal) [inline modules/src/ping.py]: spacing around +

File: modules/src/ping.py [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+import json
+
+from templates.text import TextTemplate
+
+def process(input, entitites):
+    output = {}
+    try:
+        url = entities['url'][0]['value']
+        orig_url = url
+        strip_list = ['http://','https://','/']
```
Comment (swapagarwal) [inline modules/src/ping.py]: Add a space after comma. Can you explain why the third entry in the list is required?

Comment (Wert1996) [inline modules/src/ping.py]: Coz queries like google.com/ will cause wrong domain name error.

**Comment by Wert1996** [comment]
Thanks @swapagarwal  :)

**Comment by swapagarwal** [comment]
Congratulations! 🎉 
Good work on your first pull request. 😄

**Pull request merged**